src/Display/Homepage.js

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getPosts } from '../redux/slice/postSlice';
+import Movies from '../components/Details';
+import Header from '../components/Header';
+
+import '../index.css';
+
+const Shows = () => {
+  const posts = useSelector((state) => state.post.post);
+  const dispatch = useDispatch();
+
+  const shouldGetPosts = useRef(true);
+  useEffect(() => {
+    if (shouldGetPosts.current) {
+      shouldGetPosts.current = false;
+      dispatch(getPosts());
+    }
+  }, [dispatch]);
+
+  return (
+    <div className="App">
+      <Header />
```

Comment:
- [ ] as the previous CR also mentioned you should have a way to filter your data such as you can create a search box on your homepage to filter(search) movies. if this request is not clear please leave a message in the PR or request for a 1 on 1 meeting with a senior CR.

Reply:
The search box was optional ?right?

Reply:
- Please note that you are required to add a filter function to filter the list coming from the API. For example you can filter by number of item you want to appear on the homepage or filter by alphabet. Kindly implement this function to get your PR approved 👍🏽.
![Screenshot from 2023-03-04 23-10-02](https://user-images.githubusercontent.com/44624138/222930981-3919c2d9-0463-444c-b87c-82be8c5916f9.png)

Reply:
Bt I had filtered in the single show js kindly check